src/core/atomic.d

```diff
@@ -190,6 +254,14 @@ else version( AsmX86_32 )
         //
         // +=   -=  *=  /=  %=  ^^= &=
         // |=   ^=  <<= >>= >>>=    ~=
+        static if( op == "+=" && __traits(isIntegral, T) && is(T==V1) ) {
```

Comment:
> Why is any addition going on here (and why added with mod.. isn't xadd doing this job)?

XADD returns you the value before the addition, so you have to add it again to return the total result.